packages/pwa/app/commerce-api/hooks/useCustomerProductLists.js

```diff
@@ -4,361 +4,374 @@
-            switch (action) {
-                case CustomerProductListEventQueue.eventTypes.ADD: {
-                    try {
-                        const productList = self.getProductListPerType(listType)
-                        const productListItem = productList?.customerProductListItems.find(
-                            ({productId}) => productId === event.item.id
-                        )
-                        // if the item is already in the wishlist
-                        // only update the quantity
-                        if (productListItem) {
-                            await self.updateCustomerProductListItem(productList, {
-                                ...productListItem,
-                                quantity: event.item.quantity + productListItem.quantity
-                            })
-                            eventHandler(event)
-                        } else {
-                            await self.createCustomerProductListItem(productList, {
-                                productId: event.item.id,
-                                priority: 1,
-                                quantity: parseInt(event.item.quantity),
-                                public: false,
-                                type: 'product'
-                            })
-                            eventHandler(event)
-                        }
-                    } catch (error) {
-                        errorHandler(error)
-                    }
-                    break
-                }
-
-                case CustomerProductListEventQueue.eventTypes.REMOVE:
-                    try {
-                        await self.deleteCustomerProductListItem(list, item)
-                        eventHandler(event)
-                    } catch (error) {
-                        errorHandler(error)
-                    }
-                    break
-            }
-        })
-    }, [customerProductLists])
+    const {state, actions} = useContext(CustomerProductListsContext)
 
     const self = useMemo(() => {
         return {
-            ...customerProductLists,
-            get showLoader() {
-                return isLoading
-            },
+            data: state.productLists,
 
-            get loaded() {
-                return customerProductLists?.data?.length
+            get isInitialized() {
+                return state.productLists !== undefined
             },
 
-            getProductListPerType(type) {
-                return customerProductLists?.data.find((list) => list.type === type)
+            reset() {
+                actions.reset()
             },
 
-            clearProductLists() {
-                // clear out the product lists in context
-                setCustomerProductLists({})
+            // methods that purely send network requests
+            // these methods does not alter states stored
+            // in the context
+            requests: {
```

Comment:
What is the purpose of doing for sectioning off these functions? Are we expecting that the end user will want to use the "requests" on their own outside of the stateful environment of the hook? Or is it purely stylistic/organizational?

My concern is that this hook already different that all the others with respects to using reducers, but now the hook it getting organized differently. 

Also if you looks at the hypothetical call stack for using this hook it looks like, I'm going to call something like "createList" on the hook, that will call something like "createList" but it's a request and then after that an action with the name similar to "createList" will be called. This is all a bit confusing.

Reply:
let's chat about this one